pkg/s3store/s3store.go

```diff
@@ -135,6 +135,11 @@ type S3Store struct {
 	// MaxObjectSize is the maximum size an S3 Object can have according to S3
 	// API specifications. See link above.
 	MaxObjectSize int64
+	// MaxBufferedParts is the number of additional parts that can be received from
+	// the client and stored on disk while a part is being uploaded to S3. This
+	// can help improve throughput by not blocking the client while tusd is
+	// communicating with the S3 API, which can have unpredictable latency.
+	MaxBufferedParts int
```

Comment:
The addition of MaxBufferedParts is a very good one! Only to things: Could we make it an int64, just to align with the rest of the configuration files? Also, can we set a default value for this (maybe 20?) in the New function (just as a safety net)?

Reply:
> Could we make it an int64

👍

> can we set a default value for this (maybe 20?) in the New function (just as a safety net)?

Sure. I had left it at zero (unset) because it was closest to the existing behavior, but maybe it's better to take advantage of the buffering by default.